dashboard/src/components/home/EditJwtSecretForm/EditJwtSecretForm.tsx

```diff
@@ -0,0 +1,164 @@
+import { useDialog } from '@/components/common/DialogProvider';
+import Form from '@/components/common/Form';
+import { useCurrentWorkspaceAndApplication } from '@/hooks/useCurrentWorkspaceAndApplication';
+import Button from '@/ui/v2/Button';
+import Input from '@/ui/v2/Input';
+import {
+  refetchGetAppInjectedVariablesQuery,
+  useUpdateApplicationMutation,
+} from '@/utils/__generated__/graphql';
+import { toastStyleProps } from '@/utils/settings/settingsConstants';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import * as Yup from 'yup';
+
+export interface EditJwtSecretFormProps {
+  /**
+   * Initial JWT secret.
+   */
+  jwtSecret: string;
+  /**
+   * Determines whether the form is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Submit button text.
+   *
+   * @default 'Save'
+   */
+  submitButtonText?: string;
+  /**
+   * Function to be called when the form is submitted.
+   */
+  onSubmit?: () => void;
+  /**
+   * Function to be called when the operation is cancelled.
+   */
+  onCancel?: VoidFunction;
+}
+
+export interface EditJwtSecretFormValues {
+  /**
+   * JWT secret.
+   */
+  jwtSecret: string;
+}
+
+const validationSchema = Yup.object().shape({
+  jwtSecret: Yup.string()
+    .nullable()
+    .required('This field is required.')
+    .test('isJson', 'This is not a valid JSON.', (value) => {
+      try {
+        JSON.parse(value);
+        return true;
+      } catch (error) {
+        return false;
+      }
+    }),
+});
+
+export default function EditJwtSecretForm({
```

Comment:
Hmm, how come this form is added?

Reply:
This I think is the same one we use on the Settings page.

Reply:
My bad... this one was the one I based the new form out of